### DISCUSSION TITLE
Dynamic website

### DISCUSSION THREAD

**Discussion by ShadowyNeoBTC** [sovenor/bitcoin-rocks#4] [Ideas]
Me again !

Looking at the code, I see that there is a lot of code reuse and that the code is completely static. 

Since most of the pages are similar in design and structure, we could easily use something like Angular or React to come up with a dynamic website, meaning the components are reused throughout the site.

I've done the home page locally on my computer fairly quickly.

Let me know if it's something you think is interesting.

**Comment by sovenor** [comment]
Hi again! I'm definitely interested as long as it won't impact the ability to make quick changes to the website. I really don't like working with a CMS (like WordPress for example) because it can make the website super clunky to change.

Could you share your home page code so that I can see how this would change things?

**Comment by ShadowyNeoBTC** [comment]
[src.zip](https://github.com/sovenor/bitcoin-rocks/files/14893854/src.zip)

It may seem complex if you're not used to the architecture, but it really isn't. Think of every part of the website as components. For example, you can see that each section of the homepage is defined by a definition JSON file.

I understand if this isn't the way you'd want to go, but by experience, having a lot of similar HTML files can become hard to maintain.

Let me know if you have any questions.

**Reply by sovenor** [reply]
Thanks, do you mind uploading the files into a repo so I can review them without downloading a zip?

**Reply by ShadowyNeoBTC** [reply]
Here you go !
https://github.com/L01C-CODES/bitcoin-rocks-dynamic

Look into `src/app`, that's where the code is for the homepage.

**Comment by sovenor** [comment]
Closing this discussion for now. I am happy with the current tradeoff balance struck by keeping all of the HTML pages static with dynamic jquery.i18n translations. I am open to revisiting this in the future.
